flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/SqlITCase.scala

```diff
@@ -696,6 +716,205 @@ class SqlITCase extends StreamingWithStateTestBase {
       "6,8,Hello world,51,9,5,9,1")
     assertEquals(expected.sorted, StreamITCase.testResults.sorted)
   }
+
+  @Test
+  def testAvgSumAggregatationPartition(): Unit = {
+
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    env.setParallelism(1)
+    StreamITCase.testResults = mutable.MutableList()
+
+    val sqlQuery = "SELECT a, AVG(c) OVER (PARTITION BY a ORDER BY procTime()" +
+      "RANGE BETWEEN INTERVAL '10' SECOND PRECEDING AND CURRENT ROW) AS avgC," +
+      "SUM(c) OVER (PARTITION BY a ORDER BY procTime()" +
+      "RANGE BETWEEN INTERVAL '10' SECOND PRECEDING AND CURRENT ROW) as sumC FROM MyTable"
+
+    val t = StreamTestData.get5TupleDataStream(env)
+      .toTable(tEnv).as('a, 'b, 'c, 'd, 'e)
+
+    tEnv.registerTable("MyTable", t)
+
+    val result = tEnv.sql(sqlQuery).toDataStream[Row]
+    result.addSink(new StreamITCase.StringSink)
+    env.execute()
+
+    val expected = mutable.MutableList(
+      "1,0,0",
+      "2,1,1",
+      "2,1,3",
+      "3,3,3",
+      "3,3,7",
+      "3,4,12",
+      "4,6,13",
+      "4,6,6",
+      "4,7,21",
+      "4,7,30",
+      "5,10,10",
+      "5,10,21",
+      "5,11,33",
+      "5,11,46",
+      "5,12,60")
+
+    assertEquals(expected.sorted, StreamITCase.testResults.sorted)
+  }
+
+  @Test
+  def testAvgSumAggregatationNonPartition(): Unit = {
+
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    env.setParallelism(1)
+    StreamITCase.testResults = mutable.MutableList()
+
+    val sqlQuery = "SELECT a, Count(c) OVER (ORDER BY procTime()" +
+      "RANGE BETWEEN INTERVAL '10' SECOND PRECEDING AND CURRENT ROW) AS avgC," +
+      "MIN(c) OVER (ORDER BY procTime()" +
+      "RANGE BETWEEN INTERVAL '10' SECOND PRECEDING AND CURRENT ROW) as sumC FROM MyTable"
+
+    val t = StreamTestData.get5TupleDataStream(env)
+      .toTable(tEnv).as('a, 'b, 'c, 'd, 'e)
+
+    tEnv.registerTable("MyTable", t)
+
+    val result = tEnv.sql(sqlQuery).toDataStream[Row]
+    result.addSink(new StreamITCase.StringSink)
+    env.execute()
+
+    val expected = mutable.MutableList(
+      "1,1,0",
+      "2,2,0",
+      "2,3,0",
+      "3,4,0",
+      "3,5,0",
+      "3,6,0",
+      "4,7,0",
+      "4,8,0",
+      "4,9,0",
+      "4,10,0",
+      "5,11,0",
+      "5,12,0",
+      "5,13,0",
+      "5,14,0",
+      "5,15,0")
+
+    assertEquals(expected.sorted, StreamITCase.testResults.sorted)
+  }
+
+  
+  @Test
+  def testCountAggregatationProcTimeHarnessPartitioned(): Unit = {
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
```

Comment:
Yes, the test harness allows to "control" the processing time (basically the time that this returned if you ask the process function for the current processing time).
I would move this test to a separate file though because it is a unit test (class name should end with `Test`) and is a test specifically for the `ProcTimeBoundedProcessingOverProcessFunction`.